src/scheduler.cc

```diff
@@ -296,12 +296,15 @@ Status SchedulerService::ExportFunction(ServerContext* context, const ExportFunc
   for (size_t i = 0; i < workers->size(); ++i) {
     ClientContext import_context;
     ImportFunctionRequest import_request;
-    import_request.mutable_function()->set_implementation(request->function().implementation());
+    import_request.mutable_function()->CopyFrom(request->function());
     if ((*workers)[i].current_task != ROOT_OPERATION) {
       ImportFunctionReply import_reply;
       (*workers)[i].worker_stub->ImportFunction(&import_context, import_request, &import_reply);
     }
   }
+  // TODO(rkn): Does this do a deep copy?
+  std::unique_ptr<Function> function(new Function(request->function()));
+  exported_functions.emplace_back(std::move(function));
```

Comment:
I was under the impression that it's not safe to `push_back` a `std::unique_ptr`, is that not the case?